ide/parsing.indexing/manifest.mf

```diff
@@ -4,4 +4,5 @@ OpenIDE-Module-Localizing-Bundle: org/netbeans/modules/parsing/indexing/Bundle.p
 OpenIDE-Module-Layer: org/netbeans/modules/parsing/impl/indexing/layer.xml
 AutoUpdate-Show-In-Client: false
 OpenIDE-Module-Implementation-Version: 1
+OpenIDE-Module-Needs: org.netbeans.modules.parsing.impl.indexing.implspi.ActiveDocumentProvider
```

Comment:
Just to  understand this correctly: Is the intention here, that parsing.indexing and parsing.nb are loosely coupled? I.e. parsing.indexing needs and implemenation of that SPI?

Reply:
I tried to assemble minimal application and I was getting `IllegalStateException`. Turned out that `parsing.indexing` cannot run without `ActiveDocumentProvider`. The `OpenIDE-Module-Needs` just formalizes that dependency. @tzezula may confirm, but the code in `RepositoryUpdater` speaks clearly:
```java
        this.activeDocProvider = Lookup.getDefault().lookup(ActiveDocumentProvider.class);
        if (this.activeDocProvider == null) {
            throw new IllegalStateException("No ActiveDocumentProvider instance in global lookup.");    //NOI18N
        }
```